demo.sh

```diff
@@ -0,0 +1,215 @@
+ENV_OVERRIDE_FILE=".env.override"
+NAMESPACE="opentelemetry-operator-system"
+HELM_REPO_NAME="open-telemetry"
+HELM_REPO_URL='https://open-telemetry.github.io/opentelemetry-helm-charts'
+
+DEMO_RELEASE="my-otel-demo"
+DEMO_CHART="open-telemetry/opentelemetry-demo"
+
+KUBE_STACK_RELEASE="opentelemetry-kube-stack"
+KUBE_STACK_CHART="open-telemetry/opentelemetry-kube-stack"
+KUBE_STACK_VERSION='0.9.1'
+KUBE_STACK_VALUES_URL_CLOUD='https://raw.githubusercontent.com/elastic/elastic-agent/refs/tags/v'$ELASTIC_STACK_VERSION'/deploy/helm/edot-collector/kube-stack/values.yaml'
+KUBE_STACK_VALUES_URL_SERVERLESS='https://raw.githubusercontent.com/elastic/elastic-agent/refs/tags/v'$ELASTIC_STACK_VERSION'/deploy/helm/edot-collector/kube-stack/managed_otlp/values.yaml'
+SECRET_NAME='elastic-secret-otel'
+
+DOCKER_COLLECTOR_CONFIG_CLOUD='./src/otel-collector/otelcol-elastic-config.yaml'
+DOCKER_COLLECTOR_CONFIG_SERVERLESS='./src/otel-collector/otelcol-elastic-otlp-config.yaml'
+COLLECTOR_CONTRIB_IMAGE=docker.elastic.co/elastic-agent/elastic-agent:$ELASTIC_STACK_VERSION
+
+
+# Variables
+deployment_type=""
+platform=""
+destroy="false"
+elasticsearch_endpoint=""
+elasticsearch_api_key=""
+
+usage() {
+    echo "Usage: $0 [cloud-hosted|serverless] [docker|k8s] | destroy [docker|k8s]"
+    exit 1
+}
+
+parse_args() {
+  if [ $# -eq 0 ]; then
+    usage
+  fi
+
+  if [ "$1" = "destroy" ]; then
+    destroy="true"
+    if [ $# -ge 2 ]; then
+      platform="$2"
+    fi
+    return
+  fi
+
+  deployment_type="$1"
+  if [ $# -ge 2 ]; then
+    platform="$2"
+  fi
+}
+
+update_env_var() {
+  VAR="$1"
+  VAL="$2"
+  if grep -q "^$VAR=" "$ENV_OVERRIDE_FILE"; then
+    sed -i '' "s|^$VAR=.*|$VAR=\"$VAL\"|" "$ENV_OVERRIDE_FILE"
+  else
+    echo "$VAR=\"$VAL\"" >> "$ENV_OVERRIDE_FILE"
+  fi
+}
+
+ensure_env_values() {
+  if [ -z "$elasticsearch_endpoint" ]; then
+    printf "Enter your Elastic endpoint: "
+    read elasticsearch_endpoint
+  fi
+
+  if [ -z "$elasticsearch_api_key" ]; then
+    printf "Enter your Elastic API key: "
+    read elasticsearch_api_key
+  fi
+}
+
+# Resolve OTEL Collector config path for Docker based on deployment_type
+set_docker_collector_config() {
+  case "$deployment_type" in
+    cloud-hosted)
+      OTEL_COLLECTOR_CONFIG=$DOCKER_COLLECTOR_CONFIG_CLOUD
+      ;;
+    serverless)
+      OTEL_COLLECTOR_CONFIG=$DOCKER_COLLECTOR_CONFIG_SERVERLESS
+      ;;
+  esac
+}
+
+start_docker() {
+  set_docker_collector_config
+  ensure_env_values
+
+  update_env_var "ELASTICSEARCH_ENDPOINT" "$elasticsearch_endpoint"
+  update_env_var "ELASTICSEARCH_API_KEY" "$elasticsearch_api_key"
+  update_env_var "OTEL_COLLECTOR_CONFIG" "$OTEL_COLLECTOR_CONFIG"
+  update_env_var "COLLECTOR_CONTRIB_IMAGE" "$COLLECTOR_CONTRIB_IMAGE"
+
+  make start
+}
+
+ensure_k8s_prereqs() {
+  helm repo add "$HELM_REPO_NAME" "$HELM_REPO_URL" --force-update
+  if ! kubectl get namespace "$NAMESPACE" >/dev/null 2>&1; then
+    kubectl create namespace "$NAMESPACE"
+  fi
+}
```

Comment:
> I think the script is missing creating the K8s namespace: Error from server (NotFound): error when creating "STDIN": namespaces "opentelemetry-operator-system" not found

That's odd as I created this function to check and create the namespace if it does not exist. I'll double-check

Reply:
I use a different approach in https://github.com/elastic/opentelemetry-demo/pull/160/commits/b3f02124d16fea890cbf479e304a201102378916 . Let me know if that works better